transactionparser/src/balancechanges.js

```diff
@@ -1,27 +1,50 @@
+'use strict';
+
 var _ = require('lodash');
 var BigNumber = require('bignumber.js');
 var normalizeNodes = require('./utils').normalizeNodes;
 var dropsToXRP = require('./utils').dropsToXRP;
 
-function parseBalance(balance) {
-  return new BigNumber(balance.value || balance);
+function groupByAddress(balanceChanges) {
+  var grouped = _.groupBy(balanceChanges, function(change) {
+    return change.address;
+  });
+  return _.mapValues(grouped, function(group) {
+    return _.map(group, function(change) {
+      return change.balance_change;
+    });
+  });
+}
+
+function parseValue(value) {
+  return new BigNumber(value.value || value);
 }
 
 function computeBalanceChange(node) {
+  var value = null;
+  if (node.newFields.Balance) {
+    value = parseValue(node.newFields.Balance);
+  } else if (node.previousFields.Balance && node.finalFields.Balance) {
+    value = parseValue(node.finalFields.Balance).minus(
+      parseValue(node.previousFields.Balance));
+  }
+  return value === null ? null : value.isZero() ? null : value;
+}
+
+function parseFinalBalance(node) {
   if (node.newFields.Balance) {
-    return parseBalance(node.newFields.Balance)
-  } else if(node.previousFields.Balance && node.finalFields.Balance) {
-    return parseBalance(node.finalFields.Balance).minus(
-      parseBalance(node.previousFields.Balance));
-  } else {
-    return new BigNumber(0);
+    return parseValue(node.newFields.Balance);
+  } else if (node.finalFields.Balance) {
+    return parseValue(node.finalFields.Balance);
   }
+  return null;
 }
 
-function parseXRPBalanceChange(node) {
-  var balanceChange = computeBalanceChange(node);
 
-  if(balanceChange.isZero()) {
+function parseXRPBalanceChange(node, getBalance) {
```

Comment:
We shouldn't call it "parseXRPBalanceChange" anymore because it isn't always a "change". Maybe "parseXRPQuantity"? And similar for variable names and other methods.